cmd/vcert/commands.go

```diff
@@ -613,100 +615,35 @@ func doCommandCredMgmt1(c *cli.Context) error {
-				Scope:        flags.scope,
-			})
-			if err != nil {
-				return err
-			}
-			if flags.credFormat == "json" {
-				if err := outputJSON(resp); err != nil {
-					return err
-				}
-			} else {
-				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
-				fmt.Println("access_token: ", resp.Access_token)
-				fmt.Println("access_token_expires: ", tm)
-				fmt.Println("refresh_token: ", resp.Refresh_token)
-				fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
-			}
-		} else if cfg.Credentials.User != "" && cfg.Credentials.Password != "" {
-
-			auth := &endpoint.Authentication{
-				User:     cfg.Credentials.User,
-				Password: cfg.Credentials.Password,
-				Scope:    flags.scope,
-				ClientId: flags.clientId}
-
-			if flags.sshCred {
-				auth.Scope = "ssh:manage"
-			} else if flags.pmCred {
-				auth.Scope = "certificate:manage,revoke;configuration:manage"
-			}
+	//if the email flag is provided then that means that the VaaS headless registration is the required by user
+	if flags.email != "" {
+		vaasConnector, okCasting = connector.(*cloud.Connector)
+		if !okCasting {
+			return fmt.Errorf("could not cast to VaaS connector")
+		}
+	} else {
+		tppConnector, okCasting = connector.(*tpp.Connector)
+		if !okCasting {
+			return fmt.Errorf("could not cast to TPP connector")
+		}
+	}
 
-			resp, err := tppConnector.GetRefreshToken(auth)
-			if err != nil {
-				return err
-			}
-			if flags.credFormat == "json" {
-				if err := outputJSON(resp); err != nil {
-					return err
-				}
-			} else {
-				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
-				fmt.Println("access_token: ", resp.Access_token)
-				fmt.Println("access_token_expires: ", tm)
-				if resp.Refresh_token != "" {
-					fmt.Println("refresh_token: ", resp.Refresh_token)
-					fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
-				}
-			}
-		} else if clientP12 {
-			resp, err := tppConnector.GetRefreshToken(&endpoint.Authentication{
-				ClientPKCS12: clientP12,
-				Scope:        flags.scope,
-				ClientId:     flags.clientId})
-			if err != nil {
-				return err
-			}
-			if flags.credFormat == "json" {
-				if err := outputJSON(resp); err != nil {
-					return err
-				}
-			} else {
-				tm := time.Unix(int64(resp.Expires), 0).UTC().Format(time.RFC3339)
-				fmt.Println("access_token: ", resp.Access_token)
-				fmt.Println("access_token_expires: ", tm)
-				if resp.Refresh_token != "" {
-					fmt.Println("refresh_token: ", resp.Refresh_token)
-					fmt.Println("refresh_until: ", time.Unix(int64(resp.Refresh_until), 0).UTC().Format(time.RFC3339))
-				}
-			}
+	switch c.Command.Name {
+	case commandGetCredName:
+		if vaasConnector != nil {
+			return getVaaSCredentials(vaasConnector, &cfg)
 		} else {
-			return fmt.Errorf("Failed to determine credentials set")
+			return getTppCredentials(tppConnector, &cfg, clientP12)
```

Comment:
You should check for the instantiation of the tppConnector before running the command. What if there was an error instantiating the tppConnector?

Reply:
the line 619 returns the connector if it was possible to instantiate otherwise an error is thrown, so if this part of the code is reached then we are sure that the connector was instantiate successfully.